src/reasoning/PrologClient.cpp

```diff
@@ -39,62 +42,208 @@ uint PrologClient::Instance::Get_id() const noexcept {
     return _id;
 }
 
+uint PrologClient::Instance::Get_id() const noexcept {
+    return _name;
+}
+
 
-void PrologClient::create_action_takecards(){
-    
 
-}
 
-void PrologClient::create_instance (const Instance& instance){
+void PrologClient::create_instance (const Instance& instance){ 
     PrologQueryProxy bdgs = _pl.query("rdf_costom_instance_from_class('"+ _NAMESPACE + instance._class+ ",_," +instance._name+ "', ObjInst)");
-    }
-}
+ }
 
-/*void PrologClient::create_opponent(){
-    for(i=;i<(num)=;i++){
-        Instance alfons("asdfgd","fasdfasd")M
-        create_instance(alfons);
-    }
 
-}*/
-void PrologClient::create_nao(){
-   PrologQueryProxy bdgs = _pl.query("rdf_costom_instance_from_class('" + _NAMESPACE + "Player_Nao, ObjInst')");
+
+void PrologClient::assert_property(const Instance& instance,const Property& property){
+    PrologQueryProxy bdgs = _pl.query("rdf_assert('" + _NAMESPACE + instance._class + "_" +instance._name+ "','" + _NAMESPACE + property._name + "','" + property._value + "')");
+                                                /* (instance, propertyName, propertyValue/Instance) */
 }
 
-void PrologClient::create_game(int game_num){
-    PrologQueryProxy bdgs = _pl.query("rdf_costom_instance_from_class('" + _NAMESPACE + "MemoryGame,_," +IntToStr(game_num)+ "', ObjInst)");
+void PrologClient::assert_timeStamp_property(const Instance& instance,const Instance& timeStamp){
+    PrologQueryProxy bdgs = _pl.query("rdf_assert('" + _NAMESPACE + instance._class + "_" +instance._name+ "','" + _NAMESPACE + "hasTimeStamp','" + timeStamp._name + "')");
+                                                    /*(instance, hasTimeStamp, TimeStamp_x) */
 }
 
-void PrologClient::create_ArucoToIdObjectMappings(){}
+void PrologClient::assert_CardPosition_property(const Instance& instance,const Instance& position){
+    PrologQueryProxy bdgs = _pl.query("rdf_assert('" + _NAMESPACE + instance._class + "_" +instance._name+ "','" + _NAMESPACE + "hasPositin','" + position._name + "')");
+                                                    /*(instance, hasPosition, intance_from_cardposition) */
+}
 
-void associateTurnToPlayer(const Instance& Player_instance,const Instance& Turn_instance){
-    PrologQueryProxy bdgs = _pl.query("is_in_turn('"+ _NAMESPACE + Player_instance._class+ "','"+ _NAMESPACE + Turn_instance._class+ "')");
+void PrologClient::delete_instance(const Instance& instance){
+    PrologQueryProxy bdgs = _pl.query("del('" + _NAMESPACE + instance._class+ "','" + _NAMESPACE + instance._class+  "',[])");
 
 }
 
+
 struct Position{
     unsigned int _x;
     unsigned int _y;
+    //unsigned int _time;
 }
 
 struct TurnCard {
     Position _position;
 }
 
-void save_action(const TurnCard& action){
+struct EqualCards {
```

Comment:
I think this is supposed to be an action - maybe name it like that:
e.g. TurnEqualCard